### DISCUSSION TITLE
Reporting parameters in and extracting parameter from the literature

### DISCUSSION THREAD

**Discussion by joshwlambert** [epiverse-trace/epiparameter#261] [General]
This discussion is to compile opinions and experiences from participants on building epidemiological parameter libraries, and best practises that can be documented.

**Comment by nlinton** [comment]
Hello, I was wondering how to submit via PR. The JSON file described in https://epiverse-trace.github.io/epiparameter/ is no longer at the location linked. 

![image](https://github.com/user-attachments/assets/0de4ce5a-cb54-45db-9b5c-f3c3f3445673)

**Reply by joshwlambert** [reply]
Thanks for raising this @nlinton. We are finalising the process of splitting the data from the code in {epiparameter}. The epidemiological parameter database now lives in a package called [{epiparameterDB}](https://github.com/epiverse-trace/epiparameterDB).

The process to submit new parameters in a PR would be:

**Option 1**

1. Fork and clone the {epiparameterDB} repository to get a local copy.
2. Make a new git branch (using `git checkout -b "<branch_name>"` in the terminal).
3. Add new parameters to the [`parameters.json` file](https://github.com/epiverse-trace/epiparameterDB/blob/main/inst/extdata/parameters.json). See the [data dictionary file](https://github.com/epiverse-trace/epiparameterDB/blob/main/inst/extdata/data_dictionary.json) for info on each field. This is used to automatically validate the parameter database.
4. (Optional step) Re-run the `data-raw/parameters.R` script to update the package data. (I will make sure this is done prior to merging the PR).
5. Commit and push changes to GitHub.
6. Open pull request from branch/fork into {epiparameter}, I can then merge the changes.

**Option 2**

1. Fork the repository
2. Make a new branch on GitHub
3. Edit the `parameters.json` file in GitHub and commit changes to the file
4. Open pull request from branch/fork into {epiparameter}, I can then merge the changes.

Please let me know if any of these steps are unclear and I'm happy to elaborate. We also have some general [Epiverse-TRACE contributing guidelines](https://github.com/epiverse-trace/.github/blob/main/CONTRIBUTING.md) that may help.

I will fix the link to the JSON file in the `README`.
